synapse/storage/databases/main/schema/delta/60/federation_stream_position_unique_idx.sql

```diff
@@ -0,0 +1,4 @@
+-- Ensure that only one position can be tracked for each stream type.  This might
+-- be violated if, for exapmle, a backup is restored without first clearing the
+-- table's contents.
+CREATE INDEX federation_stream_position_idx ON federation_stream_position(type);
```

Comment:
```suggestion
CREATE UNIQUE INDEX federation_stream_position_idx ON federation_stream_position(type, instance_name);
```

For sharded federation senders we expect to have multiple rows, but with different instance_names

Reply:
Ah, I made this mistake because I looked at the `full_schema` on the assumption it was the latest.  In other projects I work on, we use code generation to ensure that such things are always up-to-date, with a CI failure if that's forgotten.  Is that idea of interest?

Also, shame on me for omitting the critical "UNIQUE"!